### PR TITLE
Added landing page and file structure

### PR DESCRIPTION
**Pull request by psychoactiv**
E_COMM

- Added file structure
- Added routes
- Added shared components like nav bar, side bar and footer

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,35 @@
 import "./App.css";
-import logo from "./logo.png";
+import { Home } from "./pages/Home/Home";
+import { Cart } from "./pages/Cart/cart";
+import { ProductListing } from "./pages/ProductListing/ProductListing";
+import { Wishlist } from "./pages/Wishlist/wishlist";
+import { Login } from "./pages/Auth/login/login";
+import { Signup } from "./pages/Auth/signup/signup";
```
Comment (t007rushi) [inline src/App.js]: Put them into single  ```index.js``` and then import in single line as it's going to increase code lines into your code

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+import { Route, Routes, Link } from "react-router-dom";
+import MockMan from "mockman-js";
+import { Navbar } from "./components/navbar/navbar";
+import { StoreFooter } from "./components/Footer/footer";
+import { Sidebar } from "./components/sidebar/sidebar";
```
Comment (t007rushi) [inline src/App.js]: Same as above for components

File: src/components/Best-selling-product/best-selling.css [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  margin: 0 auto;
+}
+
+.brand-logo-container {
+  justify-content: center;
+}
+
+.all-btn-el.add-to-cart.product-card-buy {
+  padding: 0.7rem 1rem;
+  border-radius: 0px 0px 5px 5px;
```
Comment (t007rushi) [inline src/components/Best-selling-product/best-selling.css:49]: Below 1 rem can use px just suggestion and for 0px simply write 0

File: src/components/Best-selling-product/best-selling.jsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+export function BestSelling() {
+  return (
+    <Fragment>
+      <div class="best-seller-head head-font mgn-lt-4">
+        Best selling products
+      </div>
+
+      <div class="flex flex-wrap best-selling col-gap-2 row-gap-2 jc-center align-i-center">
+        <div class="product-cards flex flex-direct-col jc-sb">
+          <div class="product-display-img-container mgn-tp-0-5 mgn-rt-auto mgn-lt-auto">
```
Comment (t007rushi) [inline src/components/Best-selling-product/best-selling.jsx]: IMP Change ```class``` to ```className ```

File: src/components/BrandShoping/BrandShop.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { React, Fragment } from "react";
+import appleImg from "../../Assets/Images/58687-logo-computer-apple-icons-free-hd-image.png";
+import oneplusImg from "../../Assets/Images/OnePlus-Rebrand-Red.png";
+import samsungImg from "../../Assets/Images/Samsung-Logo.png";
```
Comment (t007rushi) [inline src/components/BrandShoping/BrandShop.jsx]: For static image put them into ```public``` folder  📁 as it will decrease imports in code for assets

**Comment by t007rushi** [comment]
Suggested some changes

**Review by t007rushi — COMMENTED** [review]
Suggested some major and some minor changes

File: src/components/Best-selling-product/best-selling.jsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            <h5 class="rubik-font mgn-tp-0-5 mgn-lt-1 product-card-head">
+              APPLE iPhone 12 (Black, 64Gb)
+            </h5>
+            <div class="mgn-tp-1 flex col-gap-1 align-i-end mgn-lt-1">
+              <span class="top-star same-star chng-pad">
+                5 <i class="fas fa-star green-star"></i>
+              </span>
+              <span class="font-wt-black card-price">₹49,000</span>
+            </div>
+            <button class="all-btn-el add-to-cart product-card-buy mgn-tp-1">
```
Comment (mariyasada) [inline src/components/Best-selling-product/best-selling.jsx]: you forgot to change class to className

File: src/components/Footer/footer.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import "./footer.css";
+export function StoreFooter() {
+  return (
+    <Fragment>
+      <footer class="e-comm-foot">
+        <div class="flex col-gap-2 flex-wrap jc-center">
+          <div class="flex flex-direct-col footer-links">
+            <div class="head-font">Socials</div>
+            <a href="#">Twitter</a>
+            <a href="#">Instagram</a>
```
Comment (mariyasada) [inline src/components/Footer/footer.jsx]: change class to className

File: src/components/banner/banner.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+          </div>
+          <div className="brand-intro-container">
+            <h3>
+              <span className="brand-intro-start rubik-font">
+                Smart phones with exciting new features and up to date Specs
+              </span>
+              <span className="brand-intro-end rubik-font">
+                {" "}
+                exclusively available only on
+              </span>
```
Comment (mariyasada) [inline src/components/banner/banner.jsx:25]: remove empty braces

File: src/contexts/sidebar-btn-context/sidebar-btn-context.jsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import { createContext, useState, useContext } from "react";
+
+const SideBarBtnContext = createContext(null);
+
+function SidebarBtnProvider({ children }) {
+  const [sideBar, setSideBar] = useState(false);
+  console.log(sideBar);
+  function toggleSidebar(e) {
```
Comment (mariyasada) [inline src/contexts/sidebar-btn-context/sidebar-btn-context.jsx]: remove consol.log

**Review by mariyasada — APPROVED** [review]
look at the comments

File: src/components/Best-selling-product/best-selling.jsx (hunk 4 above)
Comment (psychoactiv) [reply src/components/Best-selling-product/best-selling.jsx]: Sure, forgot to change that, will do it asap

File: src/components/banner/banner.jsx (hunk 8 above)
Comment (psychoactiv) [reply src/components/banner/banner.jsx:25]: Needed those braces , React was joining the tags and it made it look like single word

File: src/contexts/sidebar-btn-context/sidebar-btn-context.jsx (hunk 9 above)
Comment (psychoactiv) [reply src/contexts/sidebar-btn-context/sidebar-btn-context.jsx]: sure

File: src/components/Best-selling-product/best-selling.jsx (hunk 6 above)
Comment (psychoactiv) [reply src/components/Best-selling-product/best-selling.jsx]: Thanks, will do that

**Pull request merged**